### ISSUE TITLE
paytomany with --broadcast fails

### ISSUE THREAD

**Issue by Matoking** [spesmilo/electrum#1504]
I'm trying to create and broadcast a transaction with many outputs. At first I tried to broadcast the payment using a command similar to this

```electrum daemon start; electrum -w /home/matoking/.electrum/wallets/different_wallet paytomany --broadcast "[[\"ADDRESS1\", 0.0025], [\"ADDRESS2\", 0.02478], [\"ADDRESS3\", 0.174512]]"```

However, this fails with the following error (Electrum 2.5.1):

 ```{
    "error": "Traceback (most recent call last):\n  File \"/usr/local/bin/electrum\", line 329, in run\n    response = run_command(config, self.network, password)\n  File \"/usr/local/bin/electrum\", line 274, in run_command\n    result = func(*args)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/commands.py\", line 70, in func_wrapper\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/commands.py\", line 431, in paytomany\n    r, h = self.wallet.sendtx(tx)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/wallet.py\", line 1039, in sendtx\n    h = self.send_tx(tx)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/wallet.py\", line 1046, in send_tx\n    self.network.send([('blockchain.transaction.broadcast', [str(tx)])], self.on_broadcast)\nAttributeError: 'NoneType' object has no attribute 'send'\n" ```

So, instead of that command, which didn't work, I had to create a transaction without broadcasting it, extract the serialized transaction from the JSON data and finally broadcast it using ```electrum broadcast ```, which works the same way, even outputting the txid:

```electrum daemon start; electrum -w /home/matoking/.electrum/wallets/different_wallet paytomany  "[[\"ADDRESS1\", 0.0025], [\"ADDRESS2\", 0.02478], [\"ADDRESS3\", 0.174512]]" | jq ".hex" | xargs electrum broadcast```

**Comment by ecdsa** [comment]
what is the output of ```electrum daemon status``` ?

**Comment by ecdsa** [comment]
btw, are you running the command line as you describe above? It might be necessary to wait a bit after you start the daemon, so that it has time to create the 'network' object and establish connections.

**Comment by Matoking** [comment]
```{
    "auto_connect": true, 
    "blockchain_height": 379729, 
    "connected": true, 
    "nodes": [
        "electrum.hoodquarters.com:50002:s", 
        "46.166.165.18:50002:s", 
        "ecdsa.net:110:s", 
        "Electrum.hsmiths.com:995:s", 
        "electrum0.electricnewyear.net:50002:s", 
        "electrum1.alexcos.ro:50002:s", 
        "einfachmalnettsein.de:50002:s", 
        "anduck.net:50002:s"
    ], 
    "path": "/home/matoking/.electrum", 
    "server": "einfachmalnettsein.de", 
    "server_height": 379728
}``` 

Doesn't appear to be a problem with the lack of connection. I used a similar command in previous version to handle payments (except instead of a JSON string I pass a text file containing a CSV file of outputs, which was changed in 2.5) and it worked correctly, even so that it waits to establish a connection to an Electrum server.

**Comment by ecdsa** [comment]
I will investigate.
FYI, line 123 of ```electrum``` is supposed to ensure the network object is not None

**Comment by Matoking** [comment]
Noticed a different problem when running paytomenu --broadcast with an encrypted wallet (OldWallet). 

The program prompts for a password. If I enter the correct one I receive the following exception
```{
    "error": "Traceback (most recent call last):\n  File \"/usr/local/bin/electrum\", line 329, in run\n    response = run_command(config, self.network, password)\n  File \"/usr/local/bin/electrum\", line 274, in run_command\n    result = func(*args)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/commands.py\", line 70, in func_wrapper\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/commands.py\", line 429, in paytomany\n    tx = self._mktx(outputs, tx_fee, change_addr, domain, nocheck, unsigned)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/commands.py\", line 411, in _mktx\n    self.wallet.sign_transaction(tx, self.password)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/wallet.py\", line 1015, in sign_transaction\n    self.check_password(password)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/wallet.py\", line 1883, in check_password\n    self.accounts['0'].check_seed(seed)\n  File \"/usr/local/lib/python2.7/dist-packages/electrum/account.py\", line 245, in check_seed\n    raise InvalidPassword()\nInvalidPassword: Incorrect password\n"
}```

and if I deliberately enter a wrong password, I don't receive a JSON-encoded response but only this:
```Error: This password does not decode this wallet.```

This issue also doesn't happen if I run paywithmany *without* --broadcast.

**Comment by ecdsa** [comment]
this is because the command is passed to the daemon with --broadcast, but it fails to pass the password. Instead of fixing this, I think it makes sense to remove that option

**Comment by ecdsa** [comment]
fixed fbb65416d3072520f71ccce12a9f0acc7f29d391

you can pipe the json directly, without jq

**Comment by ecdsa** [comment]
```./electrum paytomany "[[\"15cNKqmvi9SdnY2Q11cXSzq8HnMZSonUv7\", 1]]" | ./electrum broadcast - ```
